src/main/java/appeng/client/render/cablebus/FacadeBuilder.java

```diff
@@ -19,370 +19,364 @@
+		BlockColors blockColors = Minecraft.getMinecraft().getBlockColors();
 		boolean thinFacades = isUseThinFacades( partBoxes );
 
-		CubeBuilder builder = new CubeBuilder( this.format, quads );
-
-		facadesState.forEach( ( side, state ) ->
+		for( Entry<EnumFacing, FacadeRenderState> entry : facadeStates.entrySet() )
 		{
-			AxisAlignedBB facadeBox = getFacadeBox( side, thinFacades );
-			AEAxisAlignedBB cutOutBox = getCutOutBox( facadeBox, partBoxes );
+			EnumFacing side = entry.getKey();
+			int sideIndex = side.ordinal();
+			FacadeRenderState facadeRenderState = entry.getValue();
 			boolean renderStilt = !sidesWithParts.contains( side );
-
-			try
+			if( layer == BlockRenderLayer.CUTOUT && renderStilt )
 			{
-				this.addFacade( layer, facadesState, side, cutOutBox, thinFacades, renderStilt, rand, builder );
+				for( ResourceLocation part : PartCableAnchor.FACADE_MODELS.getModels() )
+				{
+					IBakedModel partModel = modelLookup.apply( part );
+					QuadRotator rotator = new QuadRotator();
+					quads.addAll( rotator.rotateQuads( gatherQuads( partModel, null, rand ), side, EnumFacing.UP ) );
+				}
 			}
-			catch( Throwable t )
+			//If we are forcing transparency and this isn't the Translucent layer.
+			if( transparent && layer != BlockRenderLayer.TRANSLUCENT )
 			{
-				AELog.debug( t );
+				continue;
 			}
-		} );
-	}
-
-	public static TextureAtlasAndTint getSprite( IBakedModel blockModel, IBlockState state, EnumFacing facing, long rand )
-	{
 
-		TextureAtlasAndTint firstFound = null;
-		BlockRenderLayer orgLayer = MinecraftForgeClient.getRenderLayer();
-
-		try
-		{
-			// Some other mods also distinguish between layers, so we're doing this in a loop from most likely to least
-			// likely
-			for( BlockRenderLayer layer : BlockRenderLayer.values() )
+			IBlockState blockState = facadeRenderState.getSourceBlock();
+			//If we aren't forcing transparency let the block decide if it should render.
+			if( !transparent )
 			{
-
-				ForgeHooksClient.setRenderLayer( layer );
-
-				for( BakedQuad bakedQuad : blockModel.getQuads( state, facing, rand ) )
+				if( !blockState.getBlock().canRenderInLayer( blockState, layer ) )
 				{
-					return new TextureAtlasAndTint( bakedQuad );
+					continue;
 				}
+			}
 
-				for( BakedQuad bakedQuad : blockModel.getQuads( state, null, rand ) )
+			AxisAlignedBB fullBounds = thinFacades ? THIN_FACADE_BOXES[sideIndex] : THICK_FACADE_BOXES[sideIndex];
+			AxisAlignedBB facadeBox = fullBounds;
+			//If we are a transparent facade, we need to modify out BB.
+			if( facadeRenderState.isTransparent() )
+			{
+				double offset = thinFacades ? THIN_THICKNESS : THICK_THICKNESS;
+				AEAxisAlignedBB tmpBB = null;
+				for( EnumFacing face : EnumFacing.VALUES )
 				{
-					if( firstFound == null )
-					{
-						firstFound = new TextureAtlasAndTint( bakedQuad );
-					}
-					if( bakedQuad.getFace() == facing )
+					//Only faces that aren't on our axis
+					if( face.getAxis() != side.getAxis() )
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor this code to not nest more than 3 if/for/while/switch/try statements. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS134)